packages/client/src/workflow-options.ts

```diff
@@ -37,15 +37,26 @@ export interface WorkflowOptions extends CommonWorkflowOptions {
   followRuns?: boolean;
 }
 
-export interface WorkflowSignalWithStartOptions<SA extends any[] = []> extends WorkflowOptions {
+export type WorkflowSignalWithStartOptions<SignalArgs extends any[] = []> = SignalArgs extends [any, ...any[]]
+  ? WorkflowSignalWithStartOptionsWithArgs<SignalArgs>
+  : WorkflowSignalWithStartOptionsWithoutArgs;
+
+export interface WorkflowSignalWithStartOptionsWithoutArgs extends WorkflowOptions {
+  /**
+   * SignalDefinition or name of signal
+   */
+  signal: SignalDefinition | string;
```

Comment:
I'd support optional empty args here, same way we do for regular workflow `args`

Reply:
Thanks, updated